src/js/node/net.js

```diff
@@ -498,19 +579,19 @@ const Socket = (function (InternalSocket) {
     }
 
     setTimeout(timeout, callback) {
-      this.#socket?.timeout(timeout);
+      this[bunSocketInternal]?.timeout(timeout);
       this.timeout = timeout;
       if (callback) this.once("timeout", callback);
       return this;
     }
 
     unref() {
-      this.#socket?.unref();
+      this[bunSocketInternal]?.unref();
     }
 
     _write(chunk, encoding, callback) {
       if (typeof chunk == "string" && encoding !== "utf8") chunk = Buffer.from(chunk, encoding);
-      var written = this.#socket?.write(chunk);
+      var written = this[bunSocketInternal]?.write(chunk);
       if (written == chunk.length) {
```

Comment:
For non-ascii strings, this is incorrect.

Reply:
Fixed on https://github.com/oven-sh/bun/pull/3457/commits/2ff6dbd05cd2d0b7a04d48b723f98a041687e9c7